runtime/jvmti/jvmtiHook.c

```diff
@@ -1112,7 +1112,14 @@ findFieldIndexFromOffset(J9VMThread *currentThread, J9Class *clazz, UDATA offset
 		J9ROMClass * const romClass = clazz->romClass;
 		J9Class * const superclazz = GET_SUPERCLASS(clazz);
 		J9ROMFieldOffsetWalkState state;
-		J9ROMFieldOffsetWalkResult *result = vmFuncs->fieldOffsetsStartDo(vm, romClass, superclazz, &state, walkFlags);
+		J9ROMFieldOffsetWalkResult *result = NULL;
+
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		state.classLoader = clazz->classLoader;
```

Comment:
Wouldn't it be safe to just do this all the time?

Reply:
The classloader is only needed for value types. If we assign it here we will have to do the same in all the other places.

Reply:
Which we should do, since those places will fall over on value types otherwise. Why do we need the loader?